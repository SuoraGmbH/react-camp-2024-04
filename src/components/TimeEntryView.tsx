import React from "react";
import TimeEntry from "../domain/TimeEntry.ts";

interface Props {
  timeEntry: TimeEntry;
}

export const TimeEntryView: React.FunctionComponent<Props> = ({
  timeEntry,
}) => {
  if (timeEntry.id !== "1") {
    return <div>{timeEntry.comment}</div>;
  }

  return <pre>{JSON.stringify(timeEntry, undefined, 2)}</pre>;
};
