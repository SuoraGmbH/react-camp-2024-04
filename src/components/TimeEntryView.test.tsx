import { expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import { TimeEntryView } from "./TimeEntryView.tsx";
import TimeEntry from "../domain/TimeEntry.ts";

const normalTimeEntry: TimeEntry = {
  id: "5049f547-7498-4e2c-a535-29ae338428b9",
  start: new Date(),
  end: new Date(),
  comment: "This is just for tests",
};

const timeEntryWithId1: TimeEntry = {
  id: "1",
  start: new Date("2024-04-17 11:44:00"),
  end: new Date("2024-04-17 11:44:00"),
  comment: "This is for a special test case!",
};

test("ðŸ”¥", () => {
  render(<TimeEntryView timeEntry={normalTimeEntry} />);
});

it("displays the comment of the given time entry", () => {
  render(<TimeEntryView timeEntry={normalTimeEntry} />);

  expect(screen.getByText(/This is just for tests/i)).toBeVisible();
});

it("does not display the date", () => {
  render(<TimeEntryView timeEntry={normalTimeEntry} />);
  expect(screen.queryByText(/\d{4}/)).toBeNull();
});

it("for id 1 it display the debug view", () => {
  const { container } = render(<TimeEntryView timeEntry={timeEntryWithId1} />);

  expect(container).toMatchSnapshot();
});

test("1+1 is 2", () => {
  expect(1 + 2).toMatchInlineSnapshot(`3`);
});
