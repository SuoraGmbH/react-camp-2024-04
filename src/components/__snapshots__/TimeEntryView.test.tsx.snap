// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`for id 1 it display the debug view 1`] = `
<div>
  <pre>
    {
  "id": "1",
  "start": "2024-04-17T09:44:00.000Z",
  "end": "2024-04-17T09:44:00.000Z",
  "comment": "This is for a special test case!"
}
  </pre>
</div>
`;
